### PR TITLE
feat(clerk-js): Retheme <PhoneInput />

### PR DESCRIPTION
**Pull request by desiprisg**
## Description

Screenshot 2023-11-17 at 12 39 43

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Review by nikosdouvlis — APPROVED** [review]
Let's also remove the getFlagEmojiFromCountryIso helper if we no longer use it

**Comment by nikosdouvlis** [comment]
@desiprisg do we need to align the ISO codes? This is looking strange:
image

File: packages/clerk-js/src/ui.retheme/elements/PhoneInput/PhoneInput.tsx [hunk 1]
```diff
@@ -169,7 +178,13 @@ const CountryCodeListItem = memo((props: CountryCodeListItemProps) => {
       ]}
       {...rest}
     >
-      <Flag iso={country.iso} />
```
Comment (panteliselef) [inline packages/clerk-js/src/ui.retheme/elements/PhoneInput/PhoneInput.tsx:172]: Now that we removed this, you need delete the Flag component as the package is not building

**Review by panteliselef — CHANGES_REQUESTED** [review]

**Comment by desiprisg** [comment]
> @desiprisg do we need to align the ISO codes? This is looking strange: image

These will change, I was waiting for the designs!

File: packages/clerk-js/src/ui.retheme/elements/Select.tsx [hunk 2]
```diff
@@ -134,18 +140,19 @@ export const Select = withFloatingTree(<O extends Option>(props: PropsWithChildr
...
-type SelectOptionBuilderProps<O extends Option> = {
+type SelectrenderOptionProps<O extends Option> = {
   option: Option;
   index: number;
-  optionBuilder: OptionBuilder<O>;
+  renderOption: RenderOption<O>;
   handleSelect: (option: Option) => void;
-  isFocused: boolean;
+  isFocused?: boolean;
+  isSelected?: boolean;
```
Comment (panteliselef) [inline packages/clerk-js/src/ui.retheme/elements/Select.tsx:149]: These will always be passed from our component, why did we decided to make them optional ?

Comment (desiprisg) [reply packages/clerk-js/src/ui.retheme/elements/Select.tsx:149]: Component-wise, it makes more sense as they're not required.

**Review by panteliselef — APPROVED** [review]
Take a look at the comments, i think they are a handy addition

**Pull request merged**